### PR TITLE
Reduce MaxPoolV8 to MaxPoolV1 where possible

### PR DESCRIPTION
**Pull request by terfendail**
MaxPoolV8 should be inferred via ACL(V1 compatible) implementation where possible to avoid performance degradation on recent IRs that usually contain V8 even if V1 is suitable

### CODE REVIEW

File: modules/arm_plugin/src/arm_converter/arm_converter_pool.cpp [hunk 1]
```diff
@@ -69,6 +70,16 @@ template<> Converter::Conversion::Ptr Converter::Convert(const ngraph::op::v8::M
         FillLayerInfo(node, pool_info);
         pool_info.pool_type = arm_compute::PoolingType::MAX;
         return MakeConversion<arm_compute::NEPoolingLayer>(node.input(0), node.output(0), pool_info, node.output(1));
+    } else if ((node.get_input_shape(0).size() == 4) &&
```
Comment (rayonnant14) [inline modules/arm_plugin/src/arm_converter/arm_converter_pool.cpp:73]: In this case also possible to use transformation for `ngraph::op::v8::MaxPool` like `ArmPlugin::pass::ConvertMaxPool1D::ConvertMaxPool1D()` to process 3D input shapes too

Comment (terfendail) [reply modules/arm_plugin/src/arm_converter/arm_converter_pool.cpp]: I've replaced special processing with V8->V1 transformation. Now all possible cases should be handled if they are supported by V1

File: modules/arm_plugin/src/transformations/arm_optimizations.cpp [hunk 2]
```diff
@@ -277,9 +277,9 @@ bool ArmPlugin::pass::ArmOptimizations::run_on_function(std::shared_ptr<ov::Mode
         manager.register_pass<ov::pass::GraphRewrite>()->add_matcher<pass::ConvertInterpolate>();
         manager.register_pass<ov::pass::GraphRewrite>()->add_matcher<pass::ConvertMVN>();
         manager.register_pass<ov::pass::GraphRewrite>()->add_matcher<pass::ConvertReorgYolo>();
+        manager.register_pass<ov::pass::GraphRewrite>()->add_matcher<pass::ConvertMaxPoolV8>();
```
Comment (rayonnant14) [inline modules/arm_plugin/src/transformations/arm_optimizations.cpp:280]: This changes looks similar to this [transformation](https://github.com/openvinotoolkit/openvino/blob/master/src/common/transformations/src/transformations/op_conversions/convert_maxpool_downgrade.cpp). Maybe reuse it?
```cpp
manager.register_pass<ov::pass::GraphRewrite>()->add_matcher<ngraph::pass::ConvertMaxPool8ToMaxPool1>();
```

Comment (terfendail) [reply modules/arm_plugin/src/transformations/arm_optimizations.cpp]: Thanks. I missed the standard one. That's exactly what is required

**Review by rayonnant14 — APPROVED** [review]

**Pull request merged**